packages/astro/src/@types/astro.ts

```diff
@@ -849,6 +849,29 @@ export interface AstroUserConfig {
 		 * ```
 		 */
 		split?: boolean;
+
+		/**
+		 * @docs
+		 * @name build.excludeMiddleware
+		 * @type {boolean}
+		 * @default {false}
+		 * @version 2.8.0
+		 * @description
+		 * Defines how the SSR code should be bundled when built.
+		 *
+		 * When enabled, the middleware code is **not** going to be imported by the pages.
+		 *
+		 * It's going to be responsibility of the end user to execute import the code and execute it.
```

Comment:
```suggestion
		 * By default, middleware code is bundled and imported by all pages during the build. To instead execute and import middleware code manually, set `build.excludeMiddleware: true`:
```

```suggestion
		 * When enabled, middleware code is not bundled and imported by all pages during the build. To instead execute and import middleware code manually, set `build.excludeMiddleware: true`:
```
What do you think about something of this structure? I *think* prefer the one starting with default behaviour, then saying "if you want to do this instead, set true". That feels more like a definition/explanation of the config option to me, and then when to change it. I think it also helps with the "enabled means NOT" issue.  :smile:

But, something like the second could also work.